src/Operations.jl

```diff
@@ -596,22 +602,37 @@ end
 function download_artifacts(pkg_roots::Vector{String}; platform::AbstractPlatform=HostPlatform(),
                             verbose::Bool=false)
     # List of Artifacts.toml files that we're going to download from
-    artifacts_tomls = String[]
+    artifacts_tomls = Tuple{String,Dict}[]
 
     for path in pkg_roots
         # Check to see if this package has an (Julia)Artifacts.toml
         for f in artifact_names
             artifacts_toml = joinpath(path, f)
             if isfile(artifacts_toml)
-                push!(artifacts_tomls, artifacts_toml)
+                selector_path = joinpath(path, ".pkg", "select_artifacts.jl")
+
+                if isfile(selector_path)
+                    meta_toml = String(read(```
+                        $(Base.julia_cmd()) --color=no --history-file=no -O0 --startup-file=no
+                                            --project=$(path) $(selector_path)
```

Comment:
This will not work unless you also instantiate this project (which we shouldn't).

Reply:
I'm not sure why this won't work; in all my tests (such as the one I've added to `test/artifacts.jl`) it works just fine.  Can you help me to write a test case that fails?

Reply:
......huh, apparently it works locally but not on CI.  Great! :D